modules/transcription-service-ibm-watson-impl/src/main/java/org/opencastproject/transcription/ibmwatson/IBMWatsonTranscriptionService.java

```diff
@@ -206,14 +225,29 @@ public void activate(ComponentContext cc) {
       enabled = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), ENABLED_CONFIG).get();
 
       if (enabled) {
-        // User name (mandatory)
-        user = OsgiUtil.getComponentContextProperty(cc, IBM_WATSON_USER_CONFIG);
-        // Password (mandatory)
-        psw = OsgiUtil.getComponentContextProperty(cc, IBM_WATSON_PSW_CONFIG);
-        logger.info("Using transcription service at {} with username {}", IBM_WATSON_SERVICE_URL, user);
-
-        // Language model to be used
-        Option<String> modelOpt = OsgiUtil.getOptCfg(cc.getProperties(), IBM_WATSON_MODEL);
+        // Service url (optional)
+        Option<String> urlOpt = OsgiUtil.getOptCfg(cc.getProperties(), IBM_WATSON_SERVICE_URL_CONFIG);
+        if (urlOpt.isSome()) {
+          watsonServiceUrl = UrlSupport.concat(urlOpt.get(), "/", API_VERSION, "/");
```

Comment:
UrlSupport.concat handles the `/` characters for you, on both ends of the input strings.  This could be rewritten `UrlSupport.concat(urlOpt.get(), API_VERSION);`

Reply:
Thanks! I am using it in more places now (see last commit).